homeassistant/components/mqtt/schemas.py

```diff
@@ -150,3 +185,35 @@ def validate_device_has_at_least_one_identifier(value: ConfigType) -> ConfigType
         vol.Optional(CONF_UNIQUE_ID): cv.string,
     }
 )
+
+UNIQUE_ID_SCHEMA = vol.Schema(
+    {vol.Required(CONF_UNIQUE_ID): cv.string},
+).extend({}, extra=True)
+
+
+def check_unique_id(config: dict[str, Any]) -> dict[str, Any]:
+    """Check if a unique ID is set in case an entity platform is configured."""
+    platform = config[CONF_PLATFORM]
+    if platform in ENTITY_PLATFORMS and len(config.keys()) > 1:
+        UNIQUE_ID_SCHEMA(config)
+    return config
+
+
+COMPONENT_CONFIG_SCHEMA = vol.All(
```

Comment:
let's protect stuff which is not meant to be used outside this module. Same comment for other changes in this module.
```suggestion
_COMPONENT_CONFIG_SCHEMA = vol.All(
```